### PR TITLE
sourcegraph: fix and remove some default values

### PR DESCRIPTION
**Pull request by michaellzc**
### Checklist

- [ ] Follow the [manual testing process](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/TEST.md)
- [ ] Update [changelog](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/CHANGELOG.md)

### Test plan

CI

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #256** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/256" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/256?utm_source=stack-comment).

File: charts/sourcegraph/values.yaml [hunk 1]
```diff
@@ -474,8 +474,6 @@ grafana:
     readOnlyRootFilesystem: true
   # -- Name used by resources. Does not affect service names or PVCs.
   name: "grafana"
-  # -- Number of `grafana` pod
-  replicaCount: 1
```
Comment (michaellzc) [inline charts/sourcegraph/values.yaml:478]: our built-in grafana is not horizontally scable

File: charts/sourcegraph/values.yaml [hunk 2]
```diff
@@ -855,8 +853,6 @@ prometheus:
   name: "prometheus"
   # -- Enable RBAC for `prometheus`
   privileged: true
-  # -- Number of `prometheus` pod
-  replicaCount: 1
```
Comment (michaellzc) [inline charts/sourcegraph/values.yaml:859]: it should be singleton

File: charts/sourcegraph/values.yaml [hunk 3]
```diff
@@ -1093,7 +1089,7 @@ storageClass:
   parameters: {}
   # -- Persistent volumes topology configuration,
   # learn more from the [Kubernetes documentation](https://kubernetes.io/docs/concepts/storage/storage-classes/#allowed-topologies)
-  allowedTopologies: {}
+  allowedTopologies: []
```
Comment (michaellzc) [inline charts/sourcegraph/values.yaml:1092]: this is expecting an array not object

**Review by abeatrix — APPROVED** [review]
Good points! Ty!

**Review by DaedalusG — APPROVED** [review]

**Pull request merged**